editor/src/components/canvas/canvas-strategies/strategies/grid-reparent-strategy.tsx

```diff
@@ -92,120 +89,146 @@ export function gridReparentStrategy(
+      key: `draw-into-grid-strategy-controls`,
+      show: 'always-visible',
+      priority: 'bottom',
+    },
+  ]
+}
+
+export function applyGridReparent(
+  canvasState: InteractionCanvasState,
+  interactionData: DragInteractionData,
+  customStrategyState: CustomStrategyState,
+  reparentTarget: ReparentTarget,
+  selectedElements: ElementPath[],
+  gridFrame: CanvasRectangle,
+) {
+  return () => {
+    if (interactionData.drag == null) {
+      return emptyStrategyApplicationResult
+    }
+
+    const { projectContents, nodeModules } = canvasState
+    const newParent = reparentTarget.newParent
+    return ifAllowedToReparent(
+      canvasState,
+      canvasState.startingMetadata,
+      selectedElements,
+      newParent.intendedParentPath,
+      () => {
+        if (interactionData.drag == null) {
+          return emptyStrategyApplicationResult
+        }
+
+        const allowedToReparent = selectedElements.every((selectedElement) => {
+          return isAllowedToReparent(
+            canvasState.projectContents,
+            canvasState.startingMetadata,
+            selectedElement,
+            newParent.intendedParentPath,
+          )
+        })
+
+        if (!(reparentTarget.shouldReparent && allowedToReparent)) {
+          return emptyStrategyApplicationResult
+        }
+        const outcomes = mapDropNulls(
+          (selectedElement) =>
+            gridReparentCommands(
+              canvasState.startingMetadata,
+              canvasState.startingElementPathTree,
+              canvasState.startingAllElementProps,
+              canvasState.builtInDependencies,
+              projectContents,
+              nodeModules,
+              selectedElement,
+              newParent,
+            ),
+          selectedElements,
+        )
+
+        let newPaths: Array<ElementPath> = []
+        let updatedTargetPaths: UpdatedPathMap = {}
+
+        outcomes.forEach((c) => {
+          newPaths.push(c.newPath)
+          updatedTargetPaths[EP.toString(c.oldPath)] = c.newPath
+        })
+
+        const gridContainerCommands = updateBulkProperties(
+          'mid-interaction',
+          reparentTarget.newParent.intendedParentPath,
+          [
+            propertyToSet(PP.create('style', 'width'), gridFrame.width),
+            propertyToSet(PP.create('style', 'height'), gridFrame.height),
+          ],
+        )
+
+        const elementsToRerender = EP.uniqueElementPaths([
+          ...customStrategyState.elementsToRerender,
+          ...newPaths,
+          ...newPaths.map(EP.parentPath),
+          ...selectedElements.map(EP.parentPath),
+        ])
+
+        return strategyApplicationResult(
+          [
+            ...outcomes.flatMap((c) => c.commands),
+            gridContainerCommands,
+            updateSelectedViews('always', newPaths),
+            setCursorCommand(CSSCursor.Reparent),
+            showGridControls('mid-interaction', reparentTarget.newParent.intendedParentPath),
+          ],
```

Comment:
I think a setElementsToRerender command is missing from here

Reply:
it was removed on purpose in https://github.com/concrete-utopia/utopia/pull/6258 because it doesn't look right with the grid placement